### ISSUE TITLE
Fix Weird Frontend Container Image Size

### ISSUE THREAD

**Issue by zicklag** [muni-town/weird#131]
I just reallized that the Weird container image is 1.1GB which does not sound right at all. We need to investigate.

**Comment by azzamsa** [comment]
I am the author of this article https://blog.logrocket.com/optimizing-ci-cd-pipelines-rust-projects/#optimizing-docker-image-build, I will try to investigate.

**Comment by azzamsa** [comment]
Our current steps result in a whopping 1 GB image.

``` bash
docker build . -t weird:latest
```

```
✦ 🙊 docker images
REPOSITORY   TAG       IMAGE ID       CREATED          SIZE
weird        latest    2392a8f6fab2   26 seconds ago   1.1GB
```

Turns out nothing wrong with our current steps. As shown here:

```bash
weird@5b09c4e3d68a:/$ du -sh project/
3.7M    project/
```

Changing the base image solves the issue right away

```
REPOSITORY               TAG             IMAGE ID       CREATED             SIZE
weird                    latest-slim     6f135e90c544   25 minutes ago      204MB
weird                    latest-alpine   50d0765b3cf7   53 minutes ago      138MB
weird                    latest          2392a8f6fab2   About an hour ago   1.1GB
```

alpine is the smallest, but unless we have exhaustive testing, I hesitate to use it, as sometimes there is an error related to musl only. But we can try on staging for a while and see if there are any errors with the Alpine image. Otherwise, we just use the debian-slim.

---

Step to reproduce:

```bash
$ docker build . -t weird:latest
$ docker build . -t weird:latest-alpine
$ docker build . -t weird:latest-slim
```

```yaml
services:
  weird:
    image: weird:latest
    env_file: .env.local
    ports:
      - '9523:3000'
```

@zicklag what do you think? Should we give Alpine a try or just go with Debian-slim?

**Comment by azzamsa** [comment]
I can't test extensively because I get this error. `Error registering user: Error computing proof of work, you may need a browser update.` Tried to change to other web browsers but no luck.

![image](https://github.com/user-attachments/assets/7b557628-64f1-4e7f-8434-97c3b7b50efe)

**Comment by zicklag** [comment]
This was fixed in some commit around: 29e1cf124e13bdfb426851f37e17cd86c4cd9ff5
